docs/design/2019-11-05-index-advisor.md

```diff
@@ -0,0 +1,185 @@
+
+# Proposal: Index Advisor
+
+- Author(s): [Zou Huan](https://github.com/zouhuan1215) 
+- Last updated:  Nov. 5th, 2019
+- Discussion at: 
+
+## Abstract
+
+This proposal aims at designing a new DBA tool -- index advisor, which automatically recommends indexes for input workload. Having added these recommend indexes to tables, the overall execution time of input workload is supposed to decrease. Current proposal only considers single-column index recommendation.
+
+
+## Background
+
+Appropriate indexes can speed up query execution a lot. However, in real-world applications, determining proper indexes for a database according to a typical workload purely by hand, is an excessively challenging task. When deciding reasonable indexes for a large-scale database, this task becomes even more tough as one database may contain dozens of tables in which each table may have dozens of columns, whereas in the typical workload thousands of queries are involved in. Fortunately, we already have an excellent tool -- TiDB SQL optimizer that can help us get through this tough task. The main purpose of this proposal is to utilize existing TiDB SQL optimizer to automatically recommend suitable indexes for database given the workload.
+
+To illustrate how can SQL optimizer help us determine suitable indexes, here is an intuitive example: 
+
+First assume that all indexes on table `persons` are available. Then comes a query like `SELECT name FROM persons WHERE age > 18`. When responding to this query, the SQL optimizer is more likely to output an execution plan which does *index_scan* upon column `age`. The point here is: **by observing the execution plan output by SQL optimizer, we will know which indexes are favored by input queries.** 
```

Comment:
I don't understand what is the relationship between the point here and the question in previous line. Can you explain more in the docs?

Reply:
The point here is: by observing the execution plan output by SQL optimizer which requires index scan on *age*, we will know index on *age* is favored by current query. Thus, for current query, we will index on *age* instead of other columns such as *name, address, etc.*